### PR TITLE
Add Installer for OCX Microsoft Hierarchical Flex Grid

### PR DESCRIPTION
**Pull request by micahtmaclean**
Adds an installer for the OCX file required for the editing formant configurations dialog.

### CODE REVIEW

**Comment by micahtmaclean** [comment]
There might be use in writing a user guide somewhere on the installer because it won't work unless you run it as administrator.

**Comment by darcywong00** [comment]
Responding to @megahirt from #109 

> The OCX folder needs a home besides the repo root. I'm thinking DistFiles or maybe Lib. Ideas @darcywong00 ?

SpeechAnalyzer.iss should be the only file in `/Install` and it feels like `/DistFiles`  is the general dumping ground for files to get included with the installer. For this PR, I recommend moving `MSHFLXGD.OCX`, `OCXInstaller.cmd`, and `OCXInstaller.txt` to a new folder `/DistFiles/OCX/`.

Then update SpeechAnalyzer.iss to include the 3 files in the [Files] section.
I'm not sure how to get the installer to execute OCXInstaller.cmd though. That might take some digging...

**Comment by megahirt** [comment]
@darcywong00 There is a [run] section just below files where they do registration of DLLs, so I'm thinking that is where it should go.  Probably don't even need the bat file and just put the command straight in there

**Comment by micahtmaclean** [comment]
Ah, I now understand that you meant the sections in the iss file. I believe the current commit has the file section correct, but I don't understand the syntax for the run section yet. Perhaps @darcywong00 could get that working.

**Comment by darcywong00** [comment]
Maybe this [SO link](https://stackoverflow.com/questions/16429780/inno-setup-run-extracted-batch-file-as-administrator) for the [run] section

**Comment by megahirt** [comment]
> Maybe this [SO link](https://stackoverflow.com/questions/16429780/inno-setup-run-extracted-batch-file-as-administrator) for the [run] section

Yeah the SO link above looks really good - nice find.

**Comment by micahtmaclean** [comment]
Thanks @darcywong00. I've edited the ISS file to run the script. I tested it and it successfully ran the script and registered the OCX file.

 Looking at the ISS file it seems like there might be a way to do it by calling "{win}\Microsoft.NET\Framework\v4.0.30319\regasm.exe" to register it rather than the command prompt. Not certain what the difference is.

**Comment by darcywong00** [comment]
Disregard the failing automated build. I've marked it as issue #111

File: DistFiles/OCXInstaller.cmd [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cd ../
+cd File
```
Comment (darcywong00) [inline DistFiles/OCXInstaller.cmd:2]: Sorry for the delay in the review. I haven't gotten a chance to try this locally, but I was wondering why 
```
cd ../
cd File
```

Aren't `MSHFLXGD.OCX` and this .cmd file in the same folder? I'll try to get back to this PR next week...

Comment (micahtmaclean) [reply DistFiles/OCXInstaller.cmd]: Updated to fix

File: Install/SpeechAnalyzer.iss [hunk 2]
```diff
@@ -69,6 +69,8 @@ Source: "..\Lib\Fonts\Musique\MusiqueU.ttf"; DestDir: "{fonts}"; FontInstall: "M
 Source: "..\Lib\Fonts\Musique\Musique Unicode.ttf"; DestDir: "{fonts}"; FontInstall: "Musique Unicode"; Flags: onlyifdoesntexist uninsneveruninstall
 ;dtds for musicxml
 Source: "..\DistFiles\MusicXML\dtds\*" ; DestDir: "{app}\dtds"; Flags: ignoreversion
+Source: "..\DistFiles\MSHFLXGD.ocx"; DestDir: "{app}"; Flags: ignoreversion
```
Comment (darcywong00) [inline Install/SpeechAnalyzer.iss:72]: Can you insert a comment?
```
;Register MSHFlexGrid
```

**Review by darcywong00 — APPROVED** [review]
Thanks for the pull-request.
Looks good to me (can add 1 more comment to the .iss installer)

**Comment by micahtmaclean** [comment]
I've added the comments. I think the builds have been failing for some reason, which I think will prevent a merge.

**Comment by darcywong00** [comment]
>  I think the builds have been failing for some reason, which I think will prevent a merge.

Sorry about that. I made #111 to chase down the flaky CI builds, so I guess I can't procrastinate on resolving that :/

You've done everything you can for this PR, so I'll clean up the rest. Thanks

**Comment by darcywong00** [comment]
@micahtmaclean - can you merge upstream master into your fork? That should pull in the fix from #111 that fixes the failing CI build.

**Pull request merged**